src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/PreferHashDataOverComputeHash.cs

```diff
@@ -0,0 +1,298 @@
+                {
+                    switch (computeHashInvocation.Instance)
+                    {
+                        case ILocalReferenceOperation localReferenceOperation:
+                            computeHashVariableMap.TryAdd(localReferenceOperation, computeHashInvocation);
+                            break;
+                        case IInvocationOperation chainedInvocationOperation when IsHashCreateMethod(chainedInvocationOperation.TargetMethod):
+                            ReportChainedComputeHashInvocationOperation(context, computeHashInvocation, chainedInvocationOperation.TargetMethod.ContainingType, byteArrayParameter);
+                            break;
+                        case IObjectCreationOperation chainObjectCreationOperation when chainObjectCreationOperation.Type.Inherits(hashAlgoBaseType):
+                            ReportChainedComputeHashInvocationOperation(context, computeHashInvocation, chainObjectCreationOperation.Type, byteArrayParameter);
+                            break;
+                    }
+                }
+
+                void OnOperationBlockEnd(OperationBlockAnalysisContext context)
+                {
+                    var localRefCountMap = localReferenceMap.GroupBy(local => local.Value).ToDictionary(kvp => kvp.Key, kvp => kvp.Count());
+
+                    foreach (var (localReference, computeHashMethod) in computeHashVariableMap)
+                    {
+                        if (!createdSymbolMap.TryGetValue(localReference.Local, out var declarationTuple))
+                        {
+                            continue;
+                        }
+
+                        var referenceCount = localRefCountMap[localReference.Local];
+                        if (referenceCount != 1 || !TryGetHashDataMethod(declarationTuple.OriginalType, byteArrayParameter, out var staticHashMethod))
+                        {
+                            continue;
+                        }
+
+                        var codefixerLocations = GetFixerLocations(declarationTuple.DeclaratorOperation, computeHashMethod);
+                        var diagnostics = CreateDiagnostics(computeHashMethod, staticHashMethod.ContainingType, codefixerLocations);
+                        context.ReportDiagnostic(diagnostics);
+                    }
+
+                    computeHashVariableMap.Free(context.CancellationToken);
+                    createdSymbolMap.Free(context.CancellationToken);
+                    localReferenceMap.Free(context.CancellationToken);
+                }
+
+                bool IsHashCreateMethod(IMethodSymbol methodSymbol)
+                {
+                    return methodSymbol.ContainingType.Inherits(hashAlgoBaseType) &&
+                        methodSymbol.ReturnType.Inherits(hashAlgoBaseType) &&
+                        methodSymbol.Name.Equals(CreateMethodName, StringComparison.Ordinal);
```

Comment:
Should the comparison be case-insensitive for VB?

Reply:
I ran a test with `create()`. The `methodSymbol.Name` is still `Create`